src/test/scala/scorex/util/serialization/VLQReaderWriterSpecification.scala

```diff
@@ -268,4 +272,259 @@ trait VLQReaderWriterSpecification extends PropSpec
+  }
+
+  property("ULong roundtrip") {
+    forAll { v: Long => byteBufReader(byteArrayWriter().putULong(v).toBytes).getULong() shouldBe v }
+  }
+
+  property("Boolean array roundtrip") {
+    forAll { v: Array[Boolean] => byteBufReader(byteArrayWriter().putBits(v).toBytes).getBits(v.length) shouldBe v }
+  }
+
+  property("short string roundtrip") {
+    forAll(Gen.alphaStr.suchThat(_.length < 256)) { v: String =>
+      byteBufReader(byteArrayWriter().putShortString(v).toBytes).getShortString() shouldBe v
+    }
+  }
+
+  property("byte corner cases") {
+    def roundtrip(v: Byte, expected: Array[Byte]): Unit = {
+      val bytes = byteArrayWriter().put(v).toBytes
+      bytes shouldEqual expected
+      byteBufReader(expected).getByte() shouldBe v
+    }
+
+    roundtrip(Byte.MinValue, bytesFromInts(Byte.MinValue))
+    roundtrip(0, bytesFromInts(0))
+    roundtrip(Byte.MaxValue, bytesFromInts(Byte.MaxValue))
+  }
+
+  property("unsigned byte corner cases") {
+    def roundtrip(v: Int, expected: Array[Byte]): Unit = {
+      val bytes = byteArrayWriter().putUByte(v).toBytes
+      bytes shouldEqual expected
+      byteBufReader(expected).getUByte() shouldBe v
+    }
+
+    an[IllegalArgumentException] should be thrownBy roundtrip(-1, bytesFromInts(0))
+    roundtrip(0, bytesFromInts(0))
+    roundtrip(255, bytesFromInts(255))
+    an[IllegalArgumentException] should be thrownBy roundtrip(256, bytesFromInts(0))
+  }
+
+  private def prettyPrint(arr: Array[Byte]): String =
+    arr.map(b => String.format("0x%02X", Byte.box(b))).mkString(", ")
+
+  property("Short corner cases") {
+    def roundtrip(v: Short, expected: Array[Byte]): Unit = {
+      val bytes = byteArrayWriter().putShort(v).toBytes
+      withClue(s"for value $v got bytes ${prettyPrint(bytes)} (expected ${prettyPrint(expected)}): \n") {
+        bytes shouldEqual expected
+      }
+      byteBufReader(expected).getShort() shouldBe v
+    }
+
+    roundtrip(Short.MinValue, bytesFromInts(0xFF, 0xFF, 0x03))
+    roundtrip(-8193, bytesFromInts(0x81, 0x80, 0x01))
+    roundtrip(-8192, bytesFromInts(0xFF, 0x7F))
+    roundtrip(-65, bytesFromInts(0x81, 0x01))
+    roundtrip(-64, bytesFromInts(0x7F))
+    roundtrip(0, bytesFromInts(0))
+    roundtrip(63, bytesFromInts(0x7E))
+    roundtrip(64, bytesFromInts(0x80, 0x01))
+    roundtrip(8191, bytesFromInts(0xFE, 0x7F))
+    roundtrip(8192, bytesFromInts(0x80, 0x80, 0x01))
+    roundtrip(Short.MaxValue, bytesFromInts(0xFE, 0xFF, 0x03))
+  }
+
+  property("unsigned Short corner cases") {
+    def roundtrip(v: Int, expected: Array[Byte]): Unit = {
+      val bytes = byteArrayWriter().putUShort(v).toBytes
+      withClue(s"for value $v got bytes ${prettyPrint(bytes)} (expected ${prettyPrint(expected)}): \n") {
+        bytes shouldEqual expected
+      }
+      byteBufReader(expected).getUShort() shouldBe v
+    }
+
+    an[IllegalArgumentException] should be thrownBy roundtrip(-1, bytesFromInts(0))
+    roundtrip(0, bytesFromInts(0))
```

Comment:
please add at least one neighbour for each boundary case.

Reply:
Done.